pkg/imgpkg/registry/keychain.go

```diff
@@ -49,7 +50,12 @@ func (k *envKeychain) Resolve(target regauthn.Resource) (regauthn.Authenticator,
 	}
 
 	for _, info := range infos {
```

Comment:
i think we need to consider how should ordering of infos works out here e.g.

foo.com/image

vs

foo.com

i think we would want to be matching more specific cases. do we have some precedent in credential provider hows thats done?

Reply:
that makes sense... i'll add some ordering logic where the more specific match should take precedence.

Reply:
took a very similar approach to what `BasicDockerKeyring` does in terms of handling multiple urls.

https://github.com/vmware-tanzu/carvel-imgpkg/blob/3a779ca0886ab989a22c6928a1eb58cd36dc0bbd/vendor/github.com/vdemeester/k8s-pkg-credentialprovider/keyring.go#L42